### ISSUE TITLE
Problems in simulation of hydrogenation of alkynes

### ISSUE THREAD

**Issue by LucianaMB4920** [ReactionMechanismGenerator/RMG-Py#1993]
Hello!!! I am Luciana from Argentina and I'm learning to use RMG

I'm working with the hydrogenation of 1-heptyne over Ni catalyst for my PhD thesis and I am trying to simulate this reaction in RMG-cat. Just in case, I attached the paper on which I based.

I started to build my input file guided by the example of the steam reforming of methane that is on the RMG webpage but I ran into some problems.  

The species and the reaction conditions (PH2 = 1.4 bar, T = 303 K) were extracted from the paper. Also, in this paper they say that the mechanism consists of 14 reactions (with H2 dissociative or non-dissociative adsorption). In the following picture is the reaction mechanism:  

![image](https://user-images.githubusercontent.com/68957963/88813354-33ee3880-d18f-11ea-960a-001aaaae74c1.png)

Where 1-heptyne (A), 1-heptene (B) and n-heptane (C). 

Runing the input file I get in the final mode core 30 reactions, but looking at all the reactions that RMG-cat shows me in the RMG.log, there is no conversion reaction from 1-heptyne* to 1-heptene* (reactions 3 or 10 o table above) and from 1-heptyne* to n-heptane* (reactions 5 or 12) and this is what most calls my attention because they are superficial reactions. Adsorption and desorption reactions appear. I am sending my input file and the file RMG.log. I would like to add that my reactive (1-Heptyne) is liquid and I am treating it as a gas because I can not find another way. Maybe that's  the problem. 

I don't know what else to try. I tried other temperatures and pressures, but I can't get these reactions, which are the most important, since they serve to obtain the desired product (i.e. 1-heptene). 

Anyway, I wanted to ask you if you know what else I can try. Skip some species? Will I be mistreating the Adjacency List of the species? I'm losing my hopes. 

if someone could help me i will be very grateful.

Thanks in advance for reading me and have a great week! 

Regards, 
Luciana 

[inputfile1-Heptyne.zip](https://github.com/ReactionMechanismGenerator/RMG-Py/files/4995498/inputfile1-Heptyne.zip)
[RMG.log](https://github.com/ReactionMechanismGenerator/RMG-Py/files/4995499/RMG.log)
[1-Heptyne.pdf](https://github.com/ReactionMechanismGenerator/RMG-Py/files/4995511/1-Heptyne.pdf)

**Comment by cfgoldsmith** [comment]
Hi Luciana,
Can you please create a zip file with the output.html and the species subdirectory? That will make it easier for us to discuss what the mechanism is doing.
Thanks

**Comment by LucianaMB4920** [comment]
Hello Franklin!
I didn't want to bother you again but honestly, I didn't know that the github query ended up coming to you. Apologies.
I am stuck, I have made several tries but I cannot obtain the conversion reaction from 1-heptyne* to 1-heptene* (desire product) and from 1-heptyne* to n-heptane*.
In the attached input file try to write the Adjacency List of the adsorbed species in different ways. That is, as if the molecule (1-heptyne, 1-heptene, n-heptane, H and 1-heptyneH2) was linked to the site by a link and as if it were not. I don't know what the correct way is.
I tried four different conditions of pressure and temperature, which are used by the paper. Well I do not know what else to try. A shame.
I really appreciate your time and help Franklin. Hopefully it's the last time I have to bother you.
Thank you very much
Regards,
Luciana

________________________________
De: C. Franklin Goldsmith <notifications@github.com>
Enviado: jueves, 30 de julio de 2020 15:33
Para: ReactionMechanismGenerator/RMG-Py <RMG-Py@noreply.github.com>
Cc: LucianaMB4920 <luciana_mb_viale@hotmail.com>; Author <author@noreply.github.com>
Asunto: Re: [ReactionMechanismGenerator/RMG-Py] Problems in simulation of hydrogenation of alkynes (#1993)


Hi Luciana,
Can you please create a zip file with the output.html and the species subdirectory? That will make it easier for us to discuss what the mechanism is doing.
Thanks

**Comment by LucianaMB4920** [comment]
Hello Franklin!
I have made several tries but I cannot obtain the conversion reaction from 1-heptyne* to 1-heptene* (desire product) and from 1-heptyne* to n-heptane*.
In the attached input file try to write the Adjacency List of the adsorbed species in different ways. That is, as if the molecule (1-heptyne, 1-heptene, n-heptane, H and 1-heptyneH2) was linked to the site by a link and as if it were not. I don't know what the correct way is.
I tried four different conditions of pressure and temperature, which are used by the paper. Well I do not know what else to try. 
Thank you very much!!
Luciana
[Heptynefiles.zip](https://github.com/ReactionMechanismGenerator/RMG-Py/files/5003606/Heptynefiles.zip)

**Comment by cfgoldsmith** [comment]
Hi Luciana,

There are a number of issues that need to be addressed within RMG before we can properly handle hydrogenation reactions. I'm going to summarize my take on the situation. This response isn't definitive, and there are other ways that hydrogenation reactions could occur, but at the very least, we need the following steps to be fixed.

First, starting with gas-phase 1-heptyne, RMG correctly predicts that a number of dissociative adsorption reactions are possible, `A(g) + 2* --> B* + C*`. And RMG also correctly predicts that 1-heptyne can form a weakly-bound adsorbate ( or van der Waals complex, or physisorbed intermediate), `A(g) + * --> A*(vdw)`.

The problem is that neither of these reactions lead to what we need, which is a bidentate complex, i.e. an adsorbate where the two carbon atoms in the triple bond are attached individually attached to two sites: `A(g) + 2* --> A**` or `A*(vdw) + * --> A**`. Let `R` represent the C5H11 alkyl backbone in 1-heptyne. Then 1-heptyne, A, is `RC#CH`. The corresponding bidentate species should be:

```
RC = CH
 |   |
 X   X
```
So the first step is creating reaction families that are capable of generating this species. Currently that is being addressed by creating a new reaction family `vdw_to_bidentate`. That new family is being developed by kblondal. 

However, in the process of creating that family, kblondal discovered a bug in how RMG estimates the thermodynamic properties for bidentate species. This bug is summarized in issue #2003, "Bidentate adsorbates incorrectly estimated using Rj-Rj biradical not R=R molecule in gas phase". 

Next, once we have formed the initial bidentate species and have the correct thermo for it, we need for RMG to recognize that there are in fact two distinct ways of representing the same adsorbate. Specifically, when gas-phase species that contain a triple bond form a bidentate adsorbate, that adsorbate has a resonance structure. For 1-heptyne, that alternative structure is:
```
RC - CH
 ||  ||
 X   X
```
This is a known problem, see issue #2005, "Generate resonance structures for bidentate adsorbates".

Once we have this alternative structure in play, then we can more easily begin the hydrogenation process. The first reaction would be to add a hydrogen to either of the two carbon. For simplicity, I will assume it is C2, since I assume that would be more facile:
```
RC - CH      H        RHC - CH
 ||  ||   +  |   -->    |   ||   +
 X   X       X          X   X       X
```
In principle, the reaction family `Surface_Dissociation` should find that reaction (running in the reverse direction). In practice, we haven't tested in on bindenate species, so we don't know. 

Of course, the same reaction can be performed on the other resonance structure. For that reaction, we will need a new as-yet-unnamed family:
```
RC = CH    H        RHC - CH
 |  |   +  |   -->    |   ||   +
 X  X      X          X   X       X
```
Once we have formed this new product, `RHC-CH**`, then we need the reaction:

```
RHC - CH        RHC = CH
  |   ||   -->        |    +
  X   X               X        X
```
I *think* we have a reaction family under development that does this, but I am not sure.

From here, with `RHC=CH*`, it should be easy to do the reverse of `Surface_Dissociation` again to add a hydrogen to C1, which forms a vdW 1-heptene, which then desorbs to gas-phase 1-heptene. A similar sequence should be able to fully hydrogenate the alkene to an alkane.

So, in summary, there are a lot of steps/issues that need to be addressed before RMG can adequately handle hydrogenation chemistry. We are actively working on all of them, but it could take some time.

Thanks,
Franklin
